utf8/valid_support_amd64.go

```diff
@@ -0,0 +1,21 @@
+//go:build !purego
+// +build !purego
+
+package utf8
+
+import (
+	stdutf8 "unicode/utf8"
+
+	segascii "github.com/segmentio/asm/ascii"
```

Comment:
Can we go with just ascii here or is there a reason to differentiate the package name?

Reply:
That was originally the name of the return value. Renamed all returned values to either `validUtf8` or `validAscii` to clear up confusion.